django/contrib/auth/forms.py

```diff
@@ -147,9 +147,16 @@ def save(self, commit=True):
 
 
 class UserCreationForm(BaseUserCreationForm):
+    def contains_unwanted_characters(self, username):
+        """Check if the username contains characters that normalize to whitespace."""
+        normalized_username = unicodedata.normalize("NFKC", username)
+        return username != normalized_username.strip()
+
     def clean_username(self):
```

Comment:
To separate concerns, consider breaking down `clean_username()` to handle checking for unwanted characters and case for insensitive duplicates. Another request is to add unit tests for this form, covering the various scenarios.